dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -0,0 +1,698 @@
+        write_disposition: The write disposition - one of 'skip', 'append', 'replace', 'merge'.
+
+    Raises:
+        ValueError: If the write disposition is unsupported, or `id_field_name` is not
+            provided for update/merge operations.
+    """
+
+    try:
+        tbl = db_client.open_table(table_name)
+    except FileNotFoundError as e:
+        raise LanceDBBatchError(e) from e
+
+    try:
+        if write_disposition in ("append", "skip"):
+            tbl.add(records)
+        elif write_disposition == "replace":
+            tbl.add(records, mode="overwrite")
+        elif write_disposition == "merge":
+            if not id_field_name:
+                raise ValueError("To perform a merge update, 'id_field_name' must be specified.")
+            tbl.merge_insert(
+                id_field_name
+            ).when_matched_update_all().when_not_matched_insert_all().execute(records)
+        else:
+            raise ValueError(
+                f"Unsupported write disposition {write_disposition} for LanceDB Destination."
+            )
+    except ArrowInvalid as e:
+        raise LanceDBBatchError(e) from e
+    except Exception:
+        raise
+
+
+class LanceDBClient(JobClientBase, WithStateSync):
+    """LanceDB destination handler."""
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: LanceDBClientConfiguration) -> None:
+        super().__init__(schema, config)
+        self.config: LanceDBClientConfiguration = config
+        self.db_client: DBConnection = lancedb.connect(
+            uri=self.config.credentials.uri,
+            api_key=self.config.credentials.api_key,
+        )
+        self.registry = EmbeddingFunctionRegistry.get_instance()
+        self.type_mapper = LanceDBTypeMapper(self.capabilities)
+
+        # LanceDB doesn't provide a standardized way to set API keys across providers.
+        # Some use ENV variables and others allow passing api key as an argument.
+        # To account for this, we set provider environment variable as well.
+        set_non_standard_providers_environment_variables(
+            self.config.embedding_model_provider,
+            self.config.credentials.embedding_model_provider_api_key,
+        )
+        self.model_func: TextEmbeddingFunction = self.registry.get(
+            self.config.embedding_model_provider
+        ).create(
+            name=self.config.embedding_model,
+            max_retries=self.config.options.max_retries,
+            api_key=self.config.credentials.api_key,
+        )
+
+        self.vector_field_name = self.config.vector_field_name
+        self.id_field_name = self.config.id_field_name
+
+    @property
+    def dataset_name(self) -> str:
+        return self.config.normalize_dataset_name(self.schema)
+
+    @property
+    def sentinel_table(self) -> str:
+        return self.make_qualified_table_name("dltSentinelTable")
```

Comment:
i think the name of the sentinel table should be configurable, I think it is done like this in qdrant and weavieate, no?

Reply:
Doesn't seem like they are:

https://github.com/dlt-hub/dlt/blob/23c6365310d60ed86e16b25196ae4e41e71e8b3b/dlt/destinations/impl/weaviate/weaviate_client.py#L264

But I've made it configurable now: https://github.com/dlt-hub/dlt/blob/bbc00a5237f08d18a5d81c04caf6cd25bf84fe94/dlt/destinations/impl/lancedb/configuration.py#L88